scipy/stats/tests/test_distributions.py

```diff
@@ -3083,6 +3083,34 @@ def test_logcdf(self):
         assert_allclose(stats.lognorm.logsf(x2-mu, s=sigma),
                         stats.norm.logsf(np.log(x2-mu)/sigma))
 
+    @pytest.fixture(scope='function')
+    def rng(self):
+        return np.random.default_rng(1234)
+
+    @pytest.mark.parametrize("rvs_shape", [.1, 2])
+    @pytest.mark.parametrize("rvs_loc", [-2, 0, 2])
+    @pytest.mark.parametrize("rvs_scale", [.2, 1, 5])
+    @pytest.mark.parametrize('fix_shape, fix_loc, fix_scale',
+                             [e for e in product((False, True), repeat=3)
+                              if False in e])
+    @np.errstate(invalid="ignore")
```

Comment:
When do warnings occur? We would prefer to suppress warnings at the source of they are harmless.

Reply:
They used to pop up during optimisation (e.g., `RuntimeWarning: divide by zero encountered in log` when evaluating `np.log(data - loc)`); they do not appear any more, I removed the statement.